src/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.cs

```diff
@@ -62,9 +62,22 @@ public partial class MatchCollection : System.Collections.ICollection, System.Co
         void System.Collections.ICollection.CopyTo(System.Array array, int arrayIndex) { }
     }
     public delegate string MatchEvaluator(System.Text.RegularExpressions.Match match);
+    abstract public partial class RegexRunnerFactory
+    {
+        protected RegexRunnerFactory() { }
+        abstract protected internal RegexRunner CreateInstance();
+    }
     public partial class Regex
     {
         public static readonly System.TimeSpan InfiniteMatchTimeout;
+        protected internal string pattern;
```

Comment:
1. Can we put some comments in the code explaining the design?  That way the next person who comes in can understand the reasoning.
2. I don't fully understand this reasoning - we are renaming the fields in the code from `_pattern` to `pattern`. This won't work in a backwards compatible manner, will it?  If the fields are renamed, won't any existing assets need to be fixed?

Also, how come Caps and CapNames can be properties, but the rest of the values need to be fields?